ros/src/computing/perception/localization/packages/lidar_localizer/nodes/ndt_slam/ndt_slam_core.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2015-2019 Autoware Foundation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NDT_SLAM_CORE_H
+#define NDT_SLAM_CORE_H
+
+#include <array>
+#include <memory>
+#include <string>
+
+#include <ros/ros.h>
+
+#include <message_filters/subscriber.h>
+#include <message_filters/sync_policies/approximate_time.h>
+#include <message_filters/synchronizer.h>
+
+#include <tf2/transform_datatypes.h>
+#include <tf2_ros/transform_listener.h>
+#include <tf2_ros/transform_broadcaster.h>
+#include <tf2_sensor_msgs/tf2_sensor_msgs.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+
+
+#include <autoware_config_msgs/ConfigNDTSlam.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/PoseWithCovarianceStamped.h>
+#include <geometry_msgs/TwistStamped.h>
+#include <nav_msgs/Odometry.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <std_msgs/Float32.h>
+#include <std_msgs/Float64.h>
+
+#include <autoware_lidar_localizer/map_manager/map_manager.h>
+#include <autoware_lidar_localizer/matching_score/matching_score.h>
+#include <autoware_lidar_localizer/ndt/ndt_slam_pcl.h>
+#include <autoware_lidar_localizer/pose_linear_interpolator/pose_linear_interpolator.h>
+
+class NdtSlam {
+  using PointSource = pcl::PointXYZI;
+  using PointTarget = pcl::PointXYZI;
+  using SyncPolicyPoints =
+      message_filters::sync_policies::ApproximateTime<sensor_msgs::PointCloud2,
+                                                      sensor_msgs::PointCloud2>;
+
+  using SyncPolicyPointsAndPose =
+      message_filters::sync_policies::ApproximateTime<sensor_msgs::PointCloud2,
+                                                      sensor_msgs::PointCloud2,
+                                                      geometry_msgs::PoseStamped>;
+
+public:
+  NdtSlam(ros::NodeHandle nh, ros::NodeHandle private_nh);
```

Comment:
Please documentation to your methods in [doxygen](http://wiki.ros.org/rosdoc_lite) format.

Reply:
I understand that doxygen should be described because it is useful as a document.
However, there is no rule that we must write doxygen.
I'm sorry I don't have time to write all the methods, I will write in another PR.
Is it OK?